src/account/eth_account/eth_account.cairo

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo vX.Y.Z (account/eth_account/eth_account.cairo)
+
+/// # EthAccount Component
+///
+/// The EthAccount component enables contracts to behave as accounts signing with Ethereum keys.
+#[starknet::component]
+mod EthAccountComponent {
+    use core::starknet::secp256_trait::Secp256PointTrait;
+    use openzeppelin::account::eth_account::interface::EthPublicKey;
+    use openzeppelin::account::eth_account::interface;
+    use openzeppelin::account::utils::secp256k1::{Secp256k1PointSerde, Secp256k1PointStorePacking};
+    use openzeppelin::account::utils::{TRANSACTION_VERSION, QUERY_VERSION};
+    use openzeppelin::account::utils::{execute_calls, is_valid_eth_signature};
+    use openzeppelin::introspection::src5::SRC5Component::InternalTrait as SRC5InternalTrait;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use poseidon::poseidon_hash_span;
+    use starknet::account::Call;
+    use starknet::get_caller_address;
+    use starknet::get_contract_address;
+    use starknet::get_tx_info;
+
+    #[storage]
+    struct Storage {
+        EthAccount_public_key: EthPublicKey
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        OwnerAdded: OwnerAdded,
+        OwnerRemoved: OwnerRemoved
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct OwnerAdded {
+        #[key]
+        new_owner_guid: felt252
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct OwnerRemoved {
+        #[key]
+        removed_owner_guid: felt252
+    }
+
+    mod Errors {
+        const INVALID_CALLER: felt252 = 'EthAccount: invalid caller';
+        const INVALID_SIGNATURE: felt252 = 'EthAccount: invalid signature';
+        const INVALID_TX_VERSION: felt252 = 'EthAccount: invalid tx version';
+        const UNAUTHORIZED: felt252 = 'EthAccount: unauthorized';
+    }
```

Comment:
i wonder if this can be abstracted out so it's easily reused in both modules somehow

Reply:
Since we are prefixing messages with the module name, I don't think so. Do you have something in mind? We could implement traits with different implementations in scope, but that would be overkilling the issue imo.

Reply:
I was thinking more about normalizing the errors as `Account: ...` because I still feel like treating Account and EthAccount as roughly the same component, the separation wasn't ideal but a workaround. No strong opinions, we can leave it like this.